web/packages/design/src/DataTable/types.ts

```diff
@@ -105,7 +105,7 @@ export type ServersideProps = {
 
 // Makes it so either key or altKey is required
 type TableColumnWithKey<T> = TableColumnBase<T> & {
-  key: Extract<keyof T, string>;
+  key: keyof T & string;
```

Comment:
The previous version was causing errors after I changed the type passed to the request checkout table from (roughly) `PendingListItem` to  `T extends PendingListItem`. 
The error was:
<img width="910" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/534e6b3b-3587-4983-ba86-75b185ff9aa9">

I believe the new version works the same and the issue is gone.

Reply:
I'm not sure what `Extract<keyof T, string>` does. `keyof T` extracts keys from `T` and those keys themselves already are strings (I think). I tried removing it but I got lost.

Reply:
It seems that it was likely introduced due to this error you get after removing it:

```
Type 'string | number | symbol' is not assignable to type 'string'.
  Type 'number' is not assignable to type 'string'
```

But I think the solution is to clean up how those types are defined, particularly `TableColumnWithKey<T>` and `TableColumnWithAltKey<T>`.